subsys/suit/stream/stream_sinks/src/suit_ram_sink.c

```diff
@@ -110,6 +123,22 @@ static suit_plat_err_t write(void *ctx, const uint8_t *buf, size_t size)
 			memcpy(dst, buf, size);
 			ram_ctx->offset += size;
 
+			/* It is guaranteed by the suit_ram_sink_is_address_supported() that
+			 * the ram_ctx->ptr is aligned with the cache lines,
+			 * so a simple address alignment causes cache flush within the sink memory.
+			 * In case of RAM sink usage for copying data using streamers,
+			 * the destination address points to the local stack,
+			 * so flushing should not have negative effects on the system.
+			 */
+			size_t flush_size = ((uintptr_t)dst) % CACHE_ALIGNMENT;
+			int ret = sys_cache_data_flush_range(dst - flush_size, size + flush_size);
```

Comment:
`sys_cache_data_flush_range` already makes sure the whole range is covered, you don't need to do this range extension to cover full cache lines